cmd/thanos/receive.go

```diff
@@ -489,20 +497,27 @@ func runReceive(
 	}
 
 	if upload {
-		level.Debug(logger).Log("msg", "upload enabled")
-		if err := dbs.Sync(context.Background()); err != nil {
-			level.Warn(logger).Log("msg", "initial upload failed", "err", err)
+		logger := log.With(logger, "component", "uploader")
+		{
+			level.Info(logger).Log("msg", "upload enabled, starting initial sync")
+			ctx, cancel := context.WithTimeout(context.Background(), uploadTimeout)
```

Comment:
Nice, very good idea, but wonder if it should be per block or overall. 

It's hard to figure out from flag name. It looks like per block but what if someone had receiver with no upload with retention 1y and suddenly enable upload? It expects longer upload time

Reply:
I've tried to address this issue at #2908